### PR TITLE
[wasm] PackedSimd, implement store methods

### PR DESCRIPTION
**Pull request by radekdoulik**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Wasm/PackedSimd.PlatformNotSupported.cs [hunk 1]
```diff
@@ -449,6 +449,34 @@ public abstract class PackedSimd
         public static unsafe Vector128<long>   LoadWideningVector128(int*    address) { throw new PlatformNotSupportedException(); }
         public static unsafe Vector128<ulong>  LoadWideningVector128(uint*   address) { throw new PlatformNotSupportedException(); }
 
+        // Store
+
+        public static unsafe void Store(sbyte*  address, Vector128<sbyte>  source) { throw new PlatformNotSupportedException(); }
```
Comment (kg) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Wasm/PackedSimd.PlatformNotSupported.cs:454]: It's a little late to ask, but is there a specific reason why these can't be Store<T>?

File: src/mono/mono/mini/mini-llvm.c [hunk 2]
```diff
@@ -10212,6 +10210,43 @@ MONO_RESTORE_WARNING
...
+				case MONO_TYPE_U1:
+					rtype = v128_i1_t;
+					break;
+				case MONO_TYPE_I2:
+				case MONO_TYPE_U2:
+					rtype = v128_i2_t;
+					break;
+				case MONO_TYPE_R4:
+					is_fp = TRUE;
+				case MONO_TYPE_I:
```
Comment (vargaz) [inline src/mono/mono/mini/mini-llvm.c:10227]: Having these fallback cases look a bit weird, it only saves 2 lines.

**Review by vargaz — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Wasm/PackedSimd.PlatformNotSupported.cs (hunk 1 above)
Comment (radekdoulik) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Wasm/PackedSimd.PlatformNotSupported.cs:454]: It is intended to be similar to existing intrinsics API, like https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.cs#L14183

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Wasm/PackedSimd.PlatformNotSupported.cs [hunk 3]
```diff
@@ -449,6 +449,34 @@ public abstract class PackedSimd
...
+        public static unsafe void Store(uint*   address, Vector128<uint>   source) { throw new PlatformNotSupportedException(); }
+        public static unsafe void Store(long*   address, Vector128<long>   source) { throw new PlatformNotSupportedException(); }
+        public static unsafe void Store(ulong*  address, Vector128<ulong>  source) { throw new PlatformNotSupportedException(); }
+        public static unsafe void Store(float*  address, Vector128<float>  source) { throw new PlatformNotSupportedException(); }
+        public static unsafe void Store(double* address, Vector128<double> source) { throw new PlatformNotSupportedException(); }
+        public static unsafe void Store(nint*   address, Vector128<nint>   source) { throw new PlatformNotSupportedException(); }
+        public static unsafe void Store(nuint*  address, Vector128<nuint>  source) { throw new PlatformNotSupportedException(); }
+
+        public static unsafe void StoreSelectedScalar(sbyte*  address, Vector128<sbyte>  source, [ConstantExpected(Max = (byte)(15))] byte index) { throw new PlatformNotSupportedException(); } // takes ImmLaneIdx16
```
[the remark this answers is not in the document]
Comment (radekdoulik) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Wasm/PackedSimd.PlatformNotSupported.cs:467]: AdvSimd already uses this name https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.cs#L14323

**Pull request merged**